### DISCUSSION TITLE
Can't read grib dataset

### DISCUSSION THREAD

**Discussion by MTDAl** [Unidata/netcdf-java#991] [Questions and Answers] [answered]
hello, I'm trying to read grib file download from [Copernicus](https://cds.climate.copernicus.eu/) with a java program.
every time I'm trying to read grib file, I got following exception:

`

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) throws IOException {
        // TODO code application logic here
        
        String path = "C:\\Users\\*****\\1980\\01\\ERA5_19800102.grib";
        
        NetcdfDataset data = NetcdfDatasets.openDataset(path, false, null);
    }
    
`

`Exception in thread "main" java.io.IOException: Cant read C:\Users\******\1980\01\ERA5_19800103.grib: not a valid CDM file.
	at ucar.nc2.NetcdfFile.open(NetcdfFile.java:788)
	at ucar.nc2.NetcdfFile.openInMemory(NetcdfFile.java:731)
	at ucar.nc2.NetcdfFile.openInMemory(NetcdfFile.java:749)
	at com.mycompany.testnetcdf.TestNetCDF.main(TestNetCDF.java:27)`

what can I do from java code to have more information about this issue ? I use netcdf 5.4.2 
Thanks

**Comment by JohnLCaron** [comment]
Looks like you are calling NetcdfFile.openInMemory() which doesnt work on GRIB. Call NetcdfDatasets.openDataset(), which is what your code snippet claims.

If it fails, send a sample file that it fails on.

**Reply by MTDAl** [reply]
I already use openDataset() method. You can find the file here : https://filetransfer.io/data-package/1O7kCtbm#link

**Reply by MTDAl** [reply]
Additionnal information, the java project I use link only the 4 dependencies from edu.ucar : cdm-core, httpservices, netcdf-java-plateform, udunits. Maybe should I had others?

**Comment by MTDAl** [comment, marked as answer]
Ok I rebuild my maven dependencies with last version ( and netcdfAll.jar ) and it seems to work. Thanks for informations JohnLCaron

**Reply by JohnLCaron** [reply]
Yes, you need grib module and its dependencies. Thats why netcdfAll worked.
